astropy/coordinates/angle_utilities.py

```diff
@@ -123,9 +123,20 @@ def t_SIGN(t):
                 t.value = -1.0
             return t
 
+        def t_DIRECTION(t):
+            r'[NnSsEeWw]$'
+            if t.value[-1] == 'N' or t.value[-1] == 'n' or t.value[-1] == 'E' or t.value[-1] == 'e':
```

Comment:
Why look at the last character only? Given that the regex will only have on character, I think this can just be `t.value == 'N'` etc.

Also, I would write this as
```
t.value = 1 if t.value in 'NnEe' else -1
```
(Note that there is no need to test for `SsWw` since those are the only other characters that can pass the regex.)